src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -34,10 +32,9 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
         private readonly DocumentVersionCache _documentVersionCache;
         private readonly ILogger _logger;
 
-        // Maps (docURI -> (resultId -> tokens)). We cache per-doc instead of storing all tokens
-        // in one giant cache to improve colorization speeds when working with multiple files.
-        private const int MaxCachesPerDoc = 6;
-        private readonly MemoryCache<DocumentUri, MemoryCache<string, VersionedSemanticTokens>> _razorDocTokensCache = new();
+        // Caches the last few responses we send back to the client so we can avoid recomputation.
+        private const int MaxCacheSize = 5;
+        private readonly List<SemanticTokensResponse> _cachedResponses = new();
```

Comment:
This losses the per-document cache. Since we're not calculating diff's anymore it makes sense to not maintain more than one entry for the same document, but as written this would store the last 5 responses from any document, even if only the top one will be used (because the others are old versions of the document).

Not a fan of using `List` directly here and then managing it as a cache in the code. If you feel MemoryCache is too heavy for this use-case I'd say to write a `private class SmallCache` and use that.

Reply:
Oh, more specifically was there something in particular that drove the removal of `MemoryCache`?